frontend/src/components/TwoLevelDropdown.test.tsx

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { render, screen, fireEvent } from '@testing-library/react';
+import * as React from 'react';
+import { testBestPractices } from 'src/TestUtils';
+import TwoLevelDropdown, {
+  DropdownItem,
+  SelectedItem,
+  TwoLevelDropdownProps,
+} from './TwoLevelDropdown';
+
+testBestPractices();
+describe('TwoLevelDropdown', () => {
+  const title = 'Dropdown title';
+  const items: DropdownItem[] = [
+    {
+      name: 'first',
+      subItems: [
+        {
+          name: 'second',
+        },
+        {
+          name: 'third',
+          secondaryName: 'fourth',
+        },
+      ],
+    },
+    {
+      name: 'fifth',
+      subItems: [
+        {
+          name: 'sixth',
+          secondaryName: 'seventh',
+        },
+      ],
+    },
+  ];
+  const selectedItem: SelectedItem = { itemName: '', subItemName: '' };
+  const setSelectedItem: (selectedItem: SelectedItem) => void = jest.fn();
+
+  function generateProps(): TwoLevelDropdownProps {
+    return {
+      title,
+      items,
+      selectedItem,
+      setSelectedItem,
+    } as TwoLevelDropdownProps;
+  }
+
+  it('Render two-level dropdown', async () => {
+    render(<TwoLevelDropdown {...generateProps()} />);
+    screen.getByText(title);
+  });
+
+  it('Render two-level dropdown with selection', async () => {
+    const props: TwoLevelDropdownProps = generateProps();
+    props.selectedItem = {
+      itemName: 'first',
+      subItemName: 'second',
+      subItemSecondaryName: 'third',
+    } as SelectedItem;
+    render(<TwoLevelDropdown {...props} />);
+    expect(screen.queryByText(title)).toBeNull();
+    screen.getByText(/first/);
+    screen.getByText(/second/);
+    screen.getByText(/third/);
+  });
+
+  it('Two-level dropdown toggle behavior', async () => {
+    render(<TwoLevelDropdown {...generateProps()} />);
+
+    expect(screen.queryByText('first')).toBeNull();
+
+    // Display the dropdown
+    fireEvent.click(screen.getByText(title));
+    screen.getByText('first');
+
+    // Hide the dropdown
+    fireEvent.click(screen.getByText(title));
+    expect(screen.queryByText('first')).toBeNull();
+
+    // Display the dropdown again
+    fireEvent.click(screen.getByText(title));
+    screen.getByText('first');
+
+    // Click on dropdown text (not button) does not close dropdown
+    fireEvent.click(screen.getByText('first'));
+    screen.getByText('first');
+
+    // Click outside closes the dropdown
+    fireEvent.click(document.body);
+    expect(screen.queryByText('first')).toBeNull();
```

Comment:
I thought that `first` should still be displayed because the dropdown button title contains it, right?

Reply:
The dropdown button title would display a clicked element, but only the items in the second-level of the dropdown (`second`, `third > fourth`, and `sixth > seventh` in the test example dropdown) can be clicked.

Those first-level items (`first` and `fifth`) are not actually selected upon click - clicking them simply does nothing, and hovering over them displays the second-level items.

If I had selected a second level item (like `second`, as done in tests below), then `/first/` would have been present in the dropdown button title; right now however, it still just shows the original title `Dropdown title`.

Reply:
Make sense, thank you so much for testing this behavior and explanation!